inference-engine/src/gna_plugin/transformations/convert_padded2valid_conv.cpp

```diff
@@ -0,0 +1,471 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convert_padded2valid_conv.hpp"
+
+#include <memory>
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/rt_info.hpp>
+
+
+using namespace ngraph;
+using namespace op;
+
+namespace {
+struct GraphData {
+    std::shared_ptr<opset1::Convolution> conv;
+    std::shared_ptr<Transpose> leading_transpose;
+    std::shared_ptr<Transpose> trailing_transpose;
+    std::shared_ptr<opset1::MaxPool> max_pool;
+    std::shared_ptr<op::util::UnaryElementwiseArithmetic> af;
+    std::shared_ptr<Node> bias_const;
+    std::shared_ptr<Node>last_op_in_sequence_for_replacement;
+    bool disable_nhwc_to_nchw_option;
+};
+
+struct ConvData {
+    size_t input_height;
+    size_t input_width;
+    size_t input_channel_count;
+    size_t filter_height;
+    size_t filter_width;
+    size_t filter_count;
+    size_t filter_dilation_x;
+    size_t filter_dilation_y;
+    size_t filter_stride_x;
+    size_t filter_stride_y;
+    size_t pads_begin_y;
+    size_t pads_begin_x;
+    size_t pads_end_y;
+    size_t pads_end_x;
+    op::PadType padding_type;
+    size_t output_channel_count;
+    Shape output_shape;
+};
+
+bool TransposeOrderMatches(std::shared_ptr<Transpose> transpose, std::vector<int64_t> order) {
+    if (!transpose)
+        return false;
+    const Output<Node>& transpose_order = transpose->input_value(1);
+    auto transpose_order_dim = transpose_order.get_shape().size();
+
+    if (transpose_order_dim != 1 || transpose_order.get_shape()[0] != order.size())
+        return false;
+
+    auto const_with_order_values = std::dynamic_pointer_cast<opset1::Constant>(transpose_order.get_node_shared_ptr());
+    if (!const_with_order_values)
+        return false;
+
+    const int64_t* data = const_with_order_values->get_data_ptr<int64_t>();
+    if (!data)
+        return false;
+
+    for (size_t i = 0; i < order.size(); i++) {
+        if (order[i] != data[i])
+            return false;
+    }
```

Comment:
consider `std::equal`

Reply:
data is a pointer and this change would require creating a vector from this array elements, so I think it's not worth it.